### ISSUE TITLE
[Feature Request] Display Zero-width non-joiner and Zero-width joiner as their ISO keyboard symbols

### ISSUE THREAD

**Issue by mruncreative** [notepad-plus-plus/notepad-plus-plus#8530]
This may be similar #8284
https://en.wikipedia.org/wiki/Zero-width_non-joiner
https://en.wikipedia.org/wiki/Zero-width_joiner
Especially ZWNJ is useful for ligature prevention in Germanic languages like English.
Example: deafly (f and l should not connect because they are separate word parts)
Dealing with these symbols is tedious without any indicators on how many are next to each other or in the text or where they are.
This is why I propose a visual indicator for them.

**Comment by mruncreative** [comment]
Further they should be ignored in a case insensitive search because you can't search for words containing them otherwise without knowing about them.

**Comment by sasumner** [comment]
Like #8284 I'm going to tag this one as Scintilla dependent.  For display purposed, Notepad++ is totally slave to Scintilla.
I'd close this as a related-duplicate except for the search part. N++ does have control over that.

**Comment by mruncreative** [comment]
I wrote a feature request on their page after creating an account. Also mentioned the the zero width space.
https://sourceforge.net/p/scintilla/feature-requests/1367/

**Comment by mruncreative** [comment]
They referred to the character representation api.
As far as I understand from #8284 it shows a character as other characters in a black rounded box like the line endings when "show all characters" is turned on.
So I request the following black boxes to be shown:
* ZWSP for the zero width space
* ZWNJ for zero-width non-joiner
* ZWJ zero-width joiner
It may look ugly but it's better than nothing.

**Comment by sasumner** [comment]
>They referred to the character representation api.

Is anyone surprised by that? :-)

**Comment by sasumner** [comment]
@mruncreative I suggest a Pythonscript (plugin) solution to this at the moment.  Detailed discussion on it may be found here:  https://community.notepad-plus-plus.org/topic/14045/invisible-characters-unwanted

**Comment by mruncreative** [comment]
But that only works in the 32-bit version, right?

**Comment by sasumner** [comment]
>But that only works in the 32-bit version, right?

Where did you get that idea from?
Should work fine in both 32 and 64 bit.

**Comment by donho** [comment]
> Notepad++ is totally slave to Scintilla.

It's not a political correct term ;)

**Comment by Daksol** [comment]
[Link to feature request item 1367 on Scintilla site](https://sourceforge.net/p/scintilla/feature-requests/1367/)

Just saw that the feature request related to this Notepad++ item has been recently closed without any changes,

Closing comment on Scintilla  item 1367 from 2021-11-12 is
>**_Closing as it appears that character representations are sufficient to implement this._**

This is disappointing. 

#### Is there an option for Notepad++ to progress this change without any movement from Scintilla?

**Comment by sXNA** [comment]
I don't think this will be implemented there are no unicode symbols for ISO keyboard symbols.
